src/modelgauge/pipeline_runner.py

```diff
@@ -109,17 +110,28 @@ def _add_prompt_segments(self, suts, include_sink=True):
             output = CsvPromptOutput(self.output_dir() / self.output_file_name, suts)
             self.pipeline_segments.append(PromptSink(suts, output))
 
-    def _add_annotator_segments(self, annotators, include_source=True):
+    def _add_annotator_segments(self, annotators, include_source=True, ensemble=None):
         if include_source:
             input = CsvAnnotatorInput(self.input_path)
             self.pipeline_segments.append(AnnotatorSource(input))
+        if ensemble:
+            # Make sure ensemble's annotators are requested
+            missing_annotators = set(ensemble.annotators) - set(annotators.keys())
+            if missing_annotators:
+                raise ValueError(
+                    f"Ensemble annotators {missing_annotators} not found in provided annotators {set(annotators.keys())}"
+                )
         self.pipeline_segments.append(AnnotatorAssigner(annotators))
         self.pipeline_segments.append(AnnotatorWorkers(annotators, self.num_workers))
+        if ensemble:
+            self.pipeline_segments.append(EnsembleWorker(ensemble))
         output = JsonlAnnotatorOutput(self.output_dir() / self.output_file_name)
-        self.pipeline_segments.append(AnnotatorSink(annotators, output))
+        self.pipeline_segments.append(AnnotatorSink(annotators, output, ensemble=ensemble is not None))
 
     def _annotator_metadata(self):
         annotator_worker = self.pipeline_segments[-2]
+        if isinstance(annotator_worker, EnsembleWorker):
```

Comment:
I think this was already a little hacky and fragile, but I think this change makes it worse. Magic numbers are always a little sus, especially ones that don't have any domain meaning.

One way to fix this would be to iterate through the `pipeline_segments` until you find what you're looking for. Another would be express the domain meaning more clearly in the object.